packages/flutter/lib/src/painting/images.dart

```diff
@@ -157,6 +166,106 @@ class DecorationImage {
   }
 }
 
+/// The painter for a [DecorationImage].
+///
+/// To obtain a painter, call [DecorationImage.createPainter].
+///
+/// To paint, call [paint]. The `onChanged` callback passed to
+/// [DecorationImage.createPainter] will be called if the image needs to paint
+/// again (e.g. because it is animated or because it had not yet loaded the
+/// first time).
```

Comment:
nit: IIUC "or because it is loaded for the first time" would be more correct.

Reply:
I tried to clarify it by adding "first time the [paint] method was called". It might not be the first time the image is loaded (e.g. if it got evicted from the cache).